src/Rendering.hs

```diff
@@ -2,32 +2,48 @@ module Rendering where
 
 import Graphics.Gloss
 import Game
-import Data.Array
+import Data.Array 
 import Utils
 
+-- Asigna un color a cada valor de celda
+cellColor :: Int -> Color
+cellColor value = case value of
+    2    -> makeColorI 238 228 218 255   -- beige
+    4    -> makeColorI 237 224 200 255   -- light orange
+    8    -> makeColorI 242 177 121 255   -- orange
+    16   -> makeColorI 245 149 99 255    -- light red
+    32   -> makeColorI 246 124 95 255    -- red
+    64   -> makeColorI 246 94 59 255     -- strong red
+    128  -> makeColorI 237 207 114 255   -- light yellow
+    256  -> makeColorI 237 204 97 255    -- yellow
+    512  -> makeColorI 237 200 80 255    -- strong yellow
+    1024 -> makeColorI 237 197 63 255    -- light green
+    2048 -> makeColorI 237 194 46 255    -- green
+    _    -> makeColorI 205 193 180 255   -- default color (para cualquier otro valor)
+
+-- Dibuja una celda con un color basado en su valor
+drawCell :: (Int, Int) -> Cell -> Picture
+drawCell _ Empty = blank
+drawCell (y, x) (Ocuppied value) = pictures [ coloredRectangle, translatedText ]
+  where
+    cellBackgroundColor = cellColor value
+    coloredRectangle = translate (fromIntegral x * cellWidth + cellWidth / 2) (fromIntegral (n - y) * cellHeight - cellHeight / 2) $ color cellBackgroundColor $ rectangleSolid cellWidth cellHeight
+    translatedText = translate (fromIntegral x * cellWidth + cellWidth / 2 -20 ) (fromIntegral (n - y) * cellHeight - cellHeight / 2 -20) $ scale 0.5 0.5 $ boldText 1.4 $ color black $ text (show value)
+
 -- GRID DRAWING
 boardGridColor :: Color
 boardGridColor = makeColorI 0 0 0 255 
 
--- combines all the lines of the board into a single picture
--- where drawlines generate a pair of lines for each valui in the list
+-- Combina todas las líneas del tablero en una sola imagen
+-- donde drawLines genera un par de líneas para cada valor en la lista
```

Comment:
Why did you translate the comments?

Reply:
A little confusion, I commit the changes with the traductions